numba/np/arrayobj.py

```diff
@@ -3803,8 +3804,9 @@ def numpy_full_dtype_nd(context, builder, sig, args):
 
     def full(shape, value, dtype):
         arr = np.empty(shape, dtype)
-        for idx in np.ndindex(arr.shape):
-            arr[idx] = value
+        arr_flat = arr.flat
+        for idx, _ in enumerate(arr_flat):
```

Comment:
Suggest using `for idx in range(len(arr_flat))` (or similar)? There's no need to set up the more complex loop structure?

Reply:
Sure, done. Performance is identical.